### PR TITLE
ls: fix for gnu test case quote-align compatibility

### PR DESCRIPTION
**Pull request by matrixhead**
This pr tries to solve  #6554

### CODE REVIEW

File: src/uucore/src/lib/features/quoting_style.rs [hunk 1]
```diff
@@ -432,7 +434,7 @@ mod tests {
     fn check_names(name: &str, map: &[(&str, &str)]) {
         assert_eq!(
             map.iter()
-                .map(|(_, style)| escape_name(name.as_ref(), &get_style(style)))
+                .map(|(_, style)| escape_name(name.as_ref(), &get_style(style), &[]))
```
Comment (sylvestre) [inline src/uucore/src/lib/features/quoting_style.rs:437]: maybe add test when it isn't empty ?

Comment (matrixhead) [reply src/uucore/src/lib/features/quoting_style.rs:437]: I should have done that

**Comment by matrixhead** [comment]
one thing I noticed while writing tests is that when `c` quoting style is given gnu ls tries to escape ':', I couldn't understand why they're doing this, but I think I matched the behavior.

File: src/uucore/src/lib/features/quoting_style.rs [hunk 2]
```diff
@@ -791,6 +834,24 @@ mod tests {
...
+            &[
+                ("one:two", "literal"),
+                ("'one:two'", "shell"),
+                ("'one:two'", "shell-always"),
+                ("'one:two'", "shell-escape"),
+                ("'one:two'", "shell-escape-always"),
+                ("\"one\\:two\"", "c"),
+                ("one\\:two", "escape"),
+            ],
+            &[':'],
```
[the remark this answers is not in the document]
Comment (matrixhead) [reply src/uucore/src/lib/features/quoting_style.rs:854]: I don't know have any real examples other than ":" but I will add something random to check this works properly when given with multiple chars

File: src/uucore/src/lib/features/quoting_style.rs [hunk 3]
```diff
@@ -285,7 +293,15 @@ fn shell_with_escape(name: &str, quotes: Quotes) -> (String, bool) {
...
-pub fn escape_name(name: &OsStr, style: &QuotingStyle) -> String {
+/// # Arguments
+/// * `name` - The name to be escaped.
+/// * `style` - The quoting style to use for escaping the name
+/// * `additional_escaped_chars` - A list of extra characters that should be treated as escaped chars for a specific context.
+pub fn escape_name(
+    name: &OsStr,
+    style: &QuotingStyle,
+    additional_escaped_chars: &[char],
+) -> String {
```
Comment (RenjiSann) [inline src/uucore/src/lib/features/quoting_style.rs:304]: Because `escape_name` is used in many places, and its `additional_escaped_chars` argument is acually used only once, I'd like to avoid having to add `&[]` to all the other function calls.

The way I did it in #6559 was just to rename the function to `escape_name_inner`, which actually take the parameter, and `escape_name` just calls `escape_name_inner` with `&[]`.

That way, you can create `escape_dirname` which calls `escape_name_inner` with `&[':']`.

File: src/uucore/src/lib/features/quoting_style.rs [hunk 4]
```diff
@@ -285,7 +293,15 @@ fn shell_with_escape(name: &str, quotes: Quotes) -> (String, bool) {
...
 /// Escape a name according to the given quoting style.
-pub fn escape_name(name: &OsStr, style: &QuotingStyle) -> String {
+/// # Arguments
+/// * `name` - The name to be escaped.
+/// * `style` - The quoting style to use for escaping the name
+/// * `additional_escaped_chars` - A list of extra characters that should be treated as escaped chars for a specific context.
+pub fn escape_name(
+    name: &OsStr,
+    style: &QuotingStyle,
+    additional_escaped_chars: &[char],
```
Comment (RenjiSann) [inline src/uucore/src/lib/features/quoting_style.rs:303]: As a broader remark, I decided to go with a simple boolean for `dirname ?`, rather than a list of escaped characters, because I don't think there are other edge cases.

Yet that absence of other edge cases might be wrong, so I don't really know what to choose here

**Review by RenjiSann — COMMENTED** [review]
The comments I made make me slightly more confident in #6559. I added the patch to fix the `quote-align` test to it.

**Comment by matrixhead** [comment]
The test cases alone make #6559 a better PR, so I'm closing this in favor of #6559.
